src/module-rebuilder.ts

```diff
@@ -105,6 +106,47 @@ export class ModuleRebuilder {
     return args;
   }
 
+  async getSupportedNapiVersions(): Promise<number[] | undefined> {
+    const binary = (await this.packageJSONFieldWithDefault(
+      'binary',
+      {}
+    )) as Record<string, number[]>;
+
+    if ('napi_versions' in binary) {
+      return binary.napi_versions;
+    }
+
+    return undefined;
+  }
+
+  async getNapiVersion(): Promise<number | undefined> {
+    const moduleNapiVersions = await this.getSupportedNapiVersions();
+
+    if (!moduleNapiVersions) {
+      // This is not a Node-API module
+      return undefined;
+    }
+
+    const electronNapiVersion = napiVersionFromElectronVersion(this.rebuilder.electronVersion);
+    
+    if (!electronNapiVersion) {
+      throw new Error(`Native module '${this.moduleName}' requires Node-API but Electron v${this.rebuilder.electronVersion} does not support Node-API`);
+    }
+
+    // Filter out Node-API versions that are too high
+    const filteredVersions = moduleNapiVersions.filter((v) => v <= electronNapiVersion);
```

Comment:
For clarity, because two double arrows makes it kind of strange to read:
```suggestion
    const filteredVersions = moduleNapiVersions.filter((v) => (v <= electronNapiVersion));
```